docs: update plugin documentation

eikowagenknecht opened the pull request:
As discussed in #874 I would like to update the plugin documentation to make onboarding for new (advanced) users easier.

I have started with documenting the plugins and did the first three by now. I think this would be a good time for a quick feedback if this style works for the maintainers. If it's good for you, I'll continue with the rest of the plugins.

eikowagenknecht committed:
docs: first few plugins documented

aarnphm commented on docs/configuration.md:
```diff
@@ -83,3 +83,88 @@ transformers: [
 
 If you'd like to make your own plugins, read the guide on [[making plugins]] for more information.
+
+By default, Quartz is configured to use the following plugins:
```
```suggestion
By default, default plugins can be found [here](https://github.com/jackyzha0/quartz/blob/v4/quartz.config.ts)
```

Future proof ourself

eikowagenknecht replied:
Done.

aarnphm commented on docs/configuration.md:
````diff
@@ -83,3 +83,88 @@ transformers: [
+
+By default, Quartz is configured to use the following plugins:
+
+```ts
+transformers: [
+  Plugin.FrontMatter(...),
+  Plugin.CreatedModifiedDate(...),
+  Plugin.Latex(...),
+  Plugin.SyntaxHighlighting(...),
+  Plugin.ObsidianFlavoredMarkdown(...),
+  Plugin.GitHubFlavoredMarkdown(),
+  Plugin.TableOfContents(),
+  Plugin.CrawlLinks(...),
+  Plugin.Description(),
+]
+
+filters: [
+  Plugin.RemoveDrafts()
+]
+
+emitters: [
+  Plugin.AliasRedirects(),
+  Plugin.ComponentResources(...),
+  Plugin.ContentPage(),
+  Plugin.FolderPage(),
+  Plugin.TagPage(),
+  Plugin.ContentIndex(...),
+  Plugin.Assets(),
+  Plugin.Static(),
+  Plugin.NotFoundPage(),
+]
+```
````
```suggestion
```

eikowagenknecht replied:
Done.

aarnphm commented on docs/configuration.md:
````diff
@@ -83,3 +83,88 @@ transformers: [
+]
+```
+
+There are a few more that you can add to the `transformers`, `filters`, and `emitters`:
````
```suggestion
There are additional [[transformers]], [[emitters]], that you can use:
```

aarnphm commented on docs/configuration.md:
````diff
@@ -83,3 +83,88 @@ transformers: [
+]
+```
+
+### Plugin.FrontMatter
+
+This plugin uses [gray-matter](https://github.com/jonschlinkert/gray-matter) to parse the frontmatter of the file for the metadata fields `title`, `tags`, `aliases` and `cssclasses`.  See [[authoring content#Syntax]] for a description of what Quartz does with the content of these fields.
+
+Options:
+
+- `delimiters`: the delimiters to use for the frontmatter. Can have one value (e.g. `"---"`) or separate values for opening and closing delimiters (e.g. `["---", "~~~"]`). Defaults to `"---"`.
+- `language`: the language to use for parsing the frontmatter. Can be `yaml` (default) or `toml`.
+
+### Plugin.CreatedModifiedDate
+
+This plugin determines the created, modified, and published dates for a file using three potential data sources: front matter metadata, Git history, and the filesystem.
+
+If it encounters invalid date formats, it defaults to the current date.
+
+Options:
+
+- `priority`: the data sources to consult for date information. Highest priority first. Possible values are `"frontmatter"`, `"git"`, and `"filesystem"`. Defaults to `"frontmatter", "git", "filesystem"]`.
+
+### Plugin.Latex
+
+This plugin adds LaTeX rendering support. See [[Latex]] for more details.
+
+Options:
+
+- `renderEngine`: the engine to use to render LaTeX equations. Can be `"katex"` for KaTeX or `"mathjax"` for MathJax. Defaults to KaTeX.
+
+### Plugin.SyntaxHighlighting
+
+This plugin adds syntax highlighting for code blocks. See [[syntax highlighting]] for more information.
+
+#### Options
+
+- `theme`: a separate id of one of the [themes bundled with Shikiji](https://shikiji.netlify.app/themes). One for light mode and one for dark mode. Defaults to `theme: { light: "github-light", dark: "github-dark" }`.
````
I would prefer this to be a table that links to each plugins docs that we wrote, other having a centralized title here.

So that we don't have to repeat ourself.

aarnphm reviewed (commented):
initial quick bits

jackyzha0 reviewed (commented):
hey, thanks for this! i would prefer this be under `docs/tags/plugin.md` so it shows up on that page. that way, it will automatically show the list of all the plugins rather than us needing to maintain it manually

also, would be great if you could make separate pages for each plugin rather than keeping it all in the same page

eikowagenknecht committed:
docs: move plugin info

eikowagenknecht committed:
docs: move plugin docs to tag based system

eikowagenknecht committed:
docs: update latex example code snippet

eikowagenknecht committed:
docs: fix spelling of latex in title

eikowagenknecht committed:
docs: add missing linebreak

eikowagenknecht committed:
docs: remove plugin tag from feature pages

eikowagenknecht commented:
Thanks for the feedback. I restructured my documentation accordingly:

- I created a seperate page for every plugin (the first four are done, more will follow).
- I removed the plugin specific parts from the features pages (including the plugin/... tag) and replaced them with a link to the relevant plugin page(s). The reason here is that one feature (for example "Authoring Content > Syntax") sometimes uses multiple plugins, so there is no 1:1 mapping possible. Also this way the plugin list directly corresponds to what the user sees in the `plugin` config section.
- On each plugin page, the available configuration options are provided. Also with the backlinks feature, it can easily be seen, which features use the plugin.

Let me know if this works for you.

aarnphm commented on docs/authoring content.md:
```diff
@@ -34,7 +34,13 @@ Some common frontmatter fields that are natively supported by Quartz:
 - `draft`: Whether to publish the page or not. This is one way to make [[private pages|pages private]] in Quartz.
 - `date`: A string representing the day the note was published. Normally uses `YYYY-MM-DD` format.
 
+The frontmatter is parsed using the [gray-matter](https://github.com/jonschlinkert/gray-matter) library.
```
huh this feels more like a details for the frontmatter page itself?

Usually this should just be consumed by dev I think

eikowagenknecht replied:
Done.

aarnphm commented on docs/authoring content.md:
```diff
@@ -34,7 +34,13 @@ Some common frontmatter fields that are natively supported by Quartz:
 When your Quartz is at a point you're happy with, you can save your changes to GitHub.
 First, make sure you've [[setting up your GitHub repository|already setup your GitHub repository]] and then do `npx quartz sync`.
+
+## Customization
+
+Frontmatter parsing for `title`, `tags`, `aliases` and `cssclasses` is a functionality of the [[Frontmatter]] plugin. Frontmatter parsing for `date` is a functionality of the [[CreatedModifiedDate]] plugin. See the plugin pages for customization options.
```
Hmm customization should also live under the plugin page?

You can convince me otherwise.

eikowagenknecht replied:
I think we need some way to link the features to the plugins that provide these features because the configuration is described there. When I read about the Syntax for the frontmatter in `features/authoring content.md` I would want to know to what extent I can change the behaviour (described in `plugin/Frontmatter` and `plugin/CreatedModifiedDate`).

Also this creates the backlinks for the plugin pages.

I'm open to suggestions how this linking could be better worded or otherwise achieved.

aarnphm replied:
you can just create a link to that page without the details, maybe something like `See also: ... for more details` or sth like that.

But it is also not a big deal, I don't have a strong opinion on this.

jackyzha0 replied:
I think this is fine to leave as @eikowagenknecht has written it here

aarnphm replied:
sg.

aarnphm commented on docs/configuration.md:
```diff
@@ -82,4 +82,6 @@ transformers: [
-If you'd like to make your own plugins, read the guide on [[making plugins]] for more information.
+The current default plugins can be found [here](https://github.com/jackyzha0/quartz/blob/v4/quartz.config.ts). For a list of all plugins and the available configuration options, visit the [[plugin]] tag overview.
+
+If you'd like to make your own plugins, read the guide on [[making plugins]] for more information.
```
```suggestion
If you'd like to make your own plugins, read the guide on [[making plugins|making custom plugins]] for more information.
```

eikowagenknecht replied:
Done.

aarnphm commented on docs/features/Obsidian compatibility.md:
```diff
@@ -1,6 +1,4 @@
 ---
-tags:
-  - plugin/transformer
```
lets add the `features` tag for this, maybe /s/plugin/features for all of these tags under the folder?

eikowagenknecht replied:
To make sure I understand you correctly: For all feature pages that previously had a plugin/*** tag, they instead should get a feature/*** tag and the new (more technical) plugin pages get the plugin/*** tags (like I already assigned)?

aarnphm replied:
yes.

jackyzha0 replied:
agree

eikowagenknecht replied:
Done.

aarnphm commented on docs/plugins/CreatedModifiedDate.md:
```diff
@@ -0,0 +1,17 @@
+---
+title: "Plugin.CreatedModifiedDate()"
```
nit: I think this can just be `CreatedModifiedDate`.

eikowagenknecht replied:
Done.

eikowagenknecht committed:
docs: shorten titles

eikowagenknecht committed:
docs: refine wording

eikowagenknecht committed:
docs: move plugin details for frontmatter

jackyzha0 commented on docs/advanced/making plugins.md:
```diff
@@ -74,8 +74,6 @@ export const Latex: QuartzTransformerPlugin<Options> = (opts?: Options) => {
     htmlPlugins() {
       if (engine === "katex") {
-        // if you need to pass options into a plugin, you
-        // can use a tuple of [plugin, options]
```
why remove this comment?

eikowagenknecht replied:
Must have gone missing when I updated the rest of the latex example to the latest version. Restored it.

jackyzha0 commented on docs/plugins/Latex.md:
```diff
@@ -0,0 +1,13 @@
+title: "Latex"
+tags:
+  - plugin/transformer
+---
```
these should all have prose descriptions about what they do (or link back to the feature page)

eikowagenknecht replied:
Since the feature pages link here already, I added the sentence "See the backlinks for functionalities where this plugin is used." to all plugin pages. If you prefer, I can make it more explicit and include a direct link to all features that use it instead.

eikowagenknecht replied:
Added a commit where I added explicit links instead. Let me know if it needs more adjusting.

eikowagenknecht committed:
docs: add features/* tags

eikowagenknecht committed:
docs: update latex example

eikowagenknecht committed:
docs: make references more explicit

eikowagenknecht committed:
docs: add stubs for the remaining plugins

jackyzha0 reviewed (commented):
Looks good! Thanks for getting started on this, will keep following for updates

eikowagenknecht committed:
docs: more descriptions

eikowagenknecht committed:
docs: fix feature tags

eikowagenknecht committed:
docs: descriptions

eikowagenknecht committed:
docs: new plugin pages

eikowagenknecht committed:
docs: update configuration page

eikowagenknecht committed:
docs: more plugin work

eikowagenknecht committed:
docs: run prettier

eikowagenknecht commented:
Ok, I think I'm mostly done. For the `Description` plugin, I couldn't figure out, what exactly happened with the results and left a few TODOs. Maybe you can add to this.

Also I was wondering why there `quartz.config.ts` in the default config sets some values (but only a fraction of them) to their default. I added a link to the docs in the config file. Maybe we can also remove everything that is just set to the defaults of the underlying plugins? I think this would make it clearer that they're using the defaults.

E.g. `Plugin.ComponentResources({ fontOrigin: "googleFonts" }),` -> `Plugin.ComponentResources(),`.

The only two remaining configs with params would be:
- `Plugin.CreatedModifiedDate({ priority: ["frontmatter", "filesystem"] })`. This could reasonably be moved to the plugin defaults instead, so that there are no two "default" values that conflict with each other.
- `Plugin.CrawlLinks({ markdownLinkResolution: "shortest" })`. This should stay as the setup tool asks for it and then sets it.

eikowagenknecht committed:
Merge branch 'v4' into feature/docs

Signed-off-by: Eiko Wagenknecht <git@eiko-wagenknecht.de>

eikowagenknecht committed:
docs: remove comments in config file and add link to docs

eikowagenknecht committed:
docs: minor fixes

eikowagenknecht committed:
docs: run prettier

aarnphm commented:
> Also I was wondering why there `quartz.config.ts` in the default config sets some values (but only a fraction of them) to their default. 
> E.g. `Plugin.ComponentResources({ fontOrigin: "googleFonts" }),` -> `Plugin.ComponentResources(),`.

It should just show minimum default, instead of listing all options. If you want to use more you can read the code for those options. The reason for ComponenResources is that we want to surface the options for local fonts. If users want to use local font it is surfaced right there by just changing it to `local`. For example some of the options in TOC or Explorer that we didn't surface is to make the config.ts less complicated.

> * `Plugin.CreatedModifiedDate({ priority: ["frontmatter", "filesystem"] })`. This could reasonably be moved to the plugin defaults instead, so that there are no two "default" values that conflict with each other.
We shouldn't remove this, since users can add "git" as well. There are certain behaviour with this and `modifiedDate` options.

> * `Plugin.CrawlLinks({ markdownLinkResolution: "shortest" })`. This should stay as the setup tool asks for it and then sets it.

This should just keep as it. similar to the componentResources, the option is important to surface uptop to users.

aarnphm commented on docs/configuration.md:
```diff
@@ -56,30 +56,48 @@ You can think of Quartz plugins as a series of transformations over content.
+- [[tags/plugin/emitter|Emitters]] **reduce** over content (e.g. creating an RSS feed or pages that list all files with a specific tag)
 
-By adding, removing, and reordering plugins from the `tranformers`, `filters`, and `emitters` fields, you can customize the behaviour of Quartz.
+You can customise the behaviour of Quartz by adding, removing and reordering plugins in the `transformers`, `filters` and `emitters` fields.
```
Oh I also forgot to mention.

Probably worth saying here is that the order of items in the array matters, atm.

So i.e: FrontMatter should always get set first, Latex second, etc.

@jackyzha0

eikowagenknecht replied:
It's right next to this sentence:

![image](https://github.com/jackyzha0/quartz/assets/1475672/588a58dd-bbe9-46d9-9c0c-b7b05b3c911e)

aarnphm replied:
oh i didn't catch that. Can you highlight `position sensitive`, or maybe I guess the term is `location-dependent` 😄

aarnphm commented on docs/features/Mermaid diagrams.md:
```diff
@@ -1,9 +1,13 @@
 By default, Quartz will render Mermaid diagrams to match the site theme.
 
 > [!warning]
-> Wondering why Mermaid diagrams may not be showing up even if you have them enabled? You may need to reorder your plugins so that `Plugin.ObsidianFlavoredMarkdown()` is _after_ `Plugin.SyntaxHighlighting()`.
```
let's also add a tag here.

eikowagenknecht replied:
Already done (in a later commit).

aarnphm commented on docs/plugins/AliasRedirects.md:
```diff
@@ -0,0 +1,18 @@
+
+This plugin has no configuration options.
+
+## Technical Details
+
+- Category: Emitter
+- Function name: `Plugin.AliasRedirects()`.
+- Source: [`quartz/plugins/emitters/aliases.ts`](https://github.com/jackyzha0/quartz/blob/v4/quartz/plugins/emitters/aliases.ts).
```
can we rename this to `API` and remove the category, since we already have the tags for that.

Might be worth to update this to dataview at some point, once we have dataview support.

eikowagenknecht replied:
Sure, I can change that if you prefer "API".

I put in the category here because it's easier to spot next to the function name and the user needs to put the function into the right place in the config file. wdyt?

aarnphm replied:
Oh I mean the source is still helpful, not sure about the `category` and `function name`, since the page itself already is the function name, and also `Plugin.AliasRedirects` here is mainly a typescript import.

So I don't think it is helpful at all.

eikowagenknecht replied:
Changed to API but left the caterogy for now until confirmation.

aarnphm commented on docs/plugins/AliasRedirects.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin generates HTML redirect pages for aliases and permalinks defined in the frontmatter of content files, so that alternative paths correctly lead to the intended content.
```
``````suggestion
This plugin creates HTML redirect pages for aliases and permalinks defined in the frontmatter of content files.

For example, A `foo.md` has the following frontmatter

```md title="foo.md"
---
title: "Foo"
alias: 
 - "bar"
---
```

The target `host.me/bar` will be redirected to `host.me/foo`

Note that these are permanent redirect.

The emitter supports the following aliases:

- `aliases`
- `alias`

``````

aarnphm commented on docs/plugins/Assets.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin manages and emits non-Markdown assets (like images, scripts, stylesheets, etc.) from the content directory to the output directory. It ensures that all required static assets are available in the generated site.
```
```suggestion
This plugin emits all non-Markdown static assets (like images, scripts, stylesheets, etc.) from your vault content under the directory `content`.

Note that all static assets will then be accessible through its path on your generated site, i.e: `host.me/path/to/static.pdf`
```

aarnphm commented on docs/plugins/CNAME.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin generates a `CNAME` file as part of the site's output. This is needed for configuring custom domain names on platforms like GitHub Pages. The domain name is extracted from the site's base URL, which is then written to a `CNAME` file in the root of the output directory.
```
```suggestion
This plugin emits a `CNAME` record that points your subdomain to the default domain of your site. 

If you want to use a custom domain name `quartz.example.com` for the site, then this is needed.

See [[Hosting]] for more information.
```

eikowagenknecht committed:
docs: spelling

aarnphm commented on docs/plugins/ComponentResources.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin manages and emits the static resources required for the Quartz framework. This includes CSS stylesheets and JavaScript scripts that enhance the functionality and aesthetics of the generated site.
```
```suggestion
This plugin emits all static resources required to render the page like stylesheets and JavaScript required for Quartz to render your site. 
```

jackyzha0 replied:
maybe mention the `cdnCaching` option under `configuration.theme` here as well

aarnphm replied:
Something along the line of:

```prolog
`cdnCaching=false` will download the fonts to make the final build full self-container. However worth noting that it might be less efficient comparing to edge-caching for hosted cdn fonts.
```

eikowagenknecht replied:
Added a note and link th the [[configuration]] page and extended the sentence there.

aarnphm commented on docs/plugins/ContentIndex.md:
```diff
@@ -0,0 +1,24 @@
+  - plugin/emitter
+---
+
+This plugin creates a comprehensive index of the site's content, generating additional resources like a sitemap and RSS feed.
```
```suggestion
This plugin emits RSS as well as a XML site map for your site. The RSS feed allows users to subscribe to content on your site and the sitemap allows search engines to better index your site.
```

jackyzha0 replied:
It also creates a `contentIndex.json` file which is used by dynamic frontend components like search and graph

aarnphm commented on docs/plugins/ContentPage.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin is a core component of the Quartz framework. It generates the HTML pages for each piece of Markdown content. It creates the full-page layout, including headers, footers, and body content, among others.
```
```suggestion
This plugin emits and transpile all markdown pages within your vault to the static HTML page. It includes creating the full-page layout, additional components, etc. See [[architecture]] for more information.
```

eikowagenknecht commented:
Ok, got it. You want the "important" values exposed more prominently to the user.

> > * `Plugin.CreatedModifiedDate({ priority: ["frontmatter", "filesystem"] })`. This could reasonably be moved to the plugin defaults instead, so that there are no two "default" values that conflict with each other.
>
> We shouldn't remove this, since users can add "git" as well. There are certain behaviour with this and `modifiedDate` options.

Shouldn't the default in the plugin (`plugins/lastmod.ts`) then also be `["frontmatter", "filesystem"]` when that's the default a new installation gets in it's `quartz.config.ts`?

This is relevant to the discussion here, because on the plugin page, the defaults are described that the plugin sets when it doesn't get any params. So it's documented now that "git" is included in the defaults, which is technically correct but doesn't matter because it's overridden in every default installation.

aarnphm commented on docs/plugins/CrawlLinks.md:
```diff
@@ -0,0 +1,27 @@
+  - plugin/transformer
+---
+
+This plugin parses links and processes them to point to the right places. It is also needed for embedded links (like images). See [[Obsidian compatibility]] for more information.
```
```suggestion
This plugin transforms and parses all internal, external links to correct target on the generated site. See [[Obsidian compatibility]] for more information.
```

jackyzha0 replied:
imo current wording here is fine

aarnphm commented on docs/plugins/Description.md:
```diff
@@ -0,0 +1,27 @@
+  - plugin/transformer
+---
+
+This plugin generates descriptions that are used in various places:
```
generally, description is used for `<head>`. 

I think it did the same across all emitter.

aarnphm commented on quartz.config.ts:
```diff
@@ -48,19 +53,14 @@ const config: QuartzConfig = {
     transformers: [
       Plugin.FrontMatter(),
       Plugin.CreatedModifiedDate({
-        // you can add 'git' here for last modified from Git
-        // if you do rely on git for dates, ensure defaultDateType is 'modified'
         priority: ["frontmatter", "filesystem"],
       }),
       Plugin.Latex({ renderEngine: "katex" }),
       Plugin.SyntaxHighlighting({
-        // uses themes bundled with Shikiji, see https://shikiji.netlify.app/themes
         theme: {
           light: "github-light",
           dark: "github-dark",
         },
-        // set this to 'true' to use the background color of the Shikiji theme
-        // if set to 'false', will use Quartz theme colors for background
```
let's keep all the inline notes here. Still be neat to have quick discussion here instead of having users to open the docs.

Docs should just be there for anyone who wants to read it. For people who just got here the first time, it might be overwhelming to redirect them to a configuration site.

aarnphm commented on docs/plugins/TagPage.md:
```diff
@@ -0,0 +1,20 @@
+  - plugin/emitter
+---
+
+This plugin creates dedicated pages for each tag used in the content. See [[folder and tag listings]] for more information.
```
/s/creates/emits

for all `plugin/emitter`.

aarnphm commented on docs/plugins/SyntaxHighlighting.md:
```diff
@@ -0,0 +1,23 @@
+  - plugin/transformer
+---
+
+This plugin is used to add syntax highlighting to code blocks in Quartz. See [[syntax highlighting]] for more information.
```
```suggestion
This plugin converts all code-block to add support for [[syntax highlighting]] for the generated site.
```

jackyzha0 replied:
imo the current phrasing is fine here

aarnphm commented on docs/plugins/Static.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin handles the static assets. It ensures that all necessary static files, such as images, scripts, and stylesheets, are included in the final build, ready for deployment.
```
```suggestion
This plugin emits all static resources required by Quartz.

> [!important]
> This is different from [[Assets]], as these static resources is located under `quartz/static`. Whereas [[Assets]] renders all static resources under `content`.
```

aarnphm commented on docs/plugins/RemoveDrafts.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/filter
+---
+
+This plugin is designed to filter out draft content from the publishing pipeline, so only finalized content is made available. It operates by checking the `draft` flag in the frontmatter of each file. See [[private pages]] for more information.
```
```suggestion
This plugin filters out content from your content vault. This determines where a [[private pages]] can be published or not.

It checks for `draft` line within your [[Frontmatter]] to determine whether the content can be published or not.
```

eikowagenknecht replied:
Done.

aarnphm commented on docs/plugins/OxHugoFlavoredMarkdown.md:
```diff
@@ -0,0 +1,29 @@
+  - plugin/transformer
+---
+
+This plugin modifies Markdown content generated by [ox-hugo](https://github.com/kaushalmodi/ox-hugo) to make it compatible with Quartz, addressing specific formatting and syntax issues. See [[OxHugo compatibility]] for more information.
```
```suggestion
This plugin provides support for [ox-hugo](https://github.com/kaushalmodi/ox-hugo) compatibility. See [[OxHugo compatibility]] for more information.

> [!important]
> This is mutually exclusive with [[Obsidian compatibility]].
```

eikowagenknecht replied:
This warning duplicates the later warning:

> [!warning]
> While you can use this together with [[ObsidianFlavoredMarkdown]], it's not recommended because it might mutate the file in unexpected ways. Use with caution.
>
> If you use `toml` frontmatter, make sure to configure the [[Frontmatter]] plugin accordingly. See [[OxHugo compatibility]] for an example.

Which do you want to keep?

aarnphm replied:
oh I didn't realize you ahve this warning later. 

I think we should recommend users to use either or. Though they can be used together, let's not recommend it.

So I think it is fine to just say that these two are mutually exclusive?

eikowagenknecht replied:
Got it, applied the other text changes with the latest commits.

aarnphm commented on docs/plugins/ObsidianFlavoredMarkdown.md:
```diff
@@ -0,0 +1,34 @@
+  - plugin/transformer
+---
+
+This plugin extends Markdown processing to include features commonly used in Obsidian, such as enhanced link handling, callouts, mermaid diagrams, checkboxes, and more. See [[Obsidian compatibility]], [[wikilinks]], [[Mermaid diagrams]] and [[callouts]] for more information.
```
```suggestion
This plugins provide support for [[Obsidian compatibility]].
```

eikowagenknecht replied:
Done.

aarnphm commented on docs/plugins/NotFoundPage.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin is designed to generate a custom 404 (Not Found) page for the website to handle broken or non-existent URLs.
```
```suggestion
This plugin emits 404 pages for handling broken/non-existent URL.
```

eikowagenknecht replied:
Done

aarnphm reviewed (commented):
few notes.

aarnphm commented:
> Shouldn't the default in the plugin (`plugins/lastmod.ts`) then also be `["frontmatter", "filesystem"]` when that's the default a new installation gets in it's `quartz.config.ts`?

Yes this is the default, but users might want to use git for the time, so it is fine to leave it here.

> This is relevant to the discussion here, because on the plugin page, the defaults are described that the plugin sets when it doesn't get any params. So it's documented now that "git" is included in the defaults, which is technically correct but doesn't matter because it's overridden in every default installation.

It is important to still surface it in the configuration for users to know about it.

eikowagenknecht committed:
docs: update docs/plugins/AliasRedirects.md

Co-authored-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

eikowagenknecht committed:
docs: update docs/plugins/Assets.md

Co-authored-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

eikowagenknecht committed:
docs: update docs/plugins/CNAME.md

Co-authored-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

eikowagenknecht committed:
docs: update docs/plugins/Static.md

Co-authored-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

eikowagenknecht committed:
docs: update docs

aarnphm commented on docs/features/Mermaid diagrams.md:
```diff
@@ -1,9 +1,13 @@
+---
+title: "Mermaid Diagrams"
+---
```
```suggestion
title: "Mermaid Diagrams"
tags:
  - feature/transformer
---
```

aarnphm reviewed (approved):
Overall initial thumbs up from me. Thanks a lot for doing this!

Will definitely wait for Jacky to take a look as well

eikowagenknecht committed:
docs: update docs/features/Mermaid diagrams.md

Co-authored-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com>

jackyzha0 commented on docs/plugins/TableOfContents.md:
```diff
@@ -0,0 +1,26 @@
+  - plugin/transformer
+---
+
+This plugin generates a table of contents (TOC) for Markdown documents. By default it is displayed in the right sidebar. See [[table of contents]] for more information.
```
maybe add a small note for the right sidebar part about how to customize the layout (just link to layout there)

eikowagenknecht replied:
Done by reworking the feature/toc documentation instead and adding it there.

jackyzha0 commented on docs/plugins/Static.md:
```diff
@@ -0,0 +1,21 @@
+  - plugin/emitter
+---
+
+This plugin emits all static resources required by Quartz.
+
+> [!important]
+> This is different from [[Assets]], as these static resources is located under `quartz/static`. Whereas [[Assets]] renders all static resources under `content`.
```
maybe good to mention common use cases here (specifically: fonts, images that need a stable location like banners, icons, etc.)

whereas assets is more for images/videos/audio that is referenced directly through your markdown content

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/Static.md:
```diff
@@ -0,0 +1,21 @@
+> [!note]
+> For information on how to add, remove or configure plugins, see the [[Configuration#Plugins|Configuration]] page.
+
+This plugin has no configuration options.
```
do note that this respects `ignorePatterns` in the global configuration

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/RemoveDrafts.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/filter
+---
+
+This plugin filters out content from your vault, so that only finalized content is made available. This prevents [[private pages]] from being published.
```
```suggestion
This plugin filters out content from your vault, so that only finalized content is made available. This prevents [[private pages]] from being published. By default, it filters out all pages with `draft: true` in the frontmatter and leaves all other pages intact.
```

jackyzha0 commented on docs/plugins/Latex.md:
```diff
@@ -0,0 +1,20 @@
+
+This plugin accepts the following configuration options:
+
+- `renderEngine`: the engine to use to render LaTeX equations. Can be `"katex"` for KaTeX or `"mathjax"` for [MathJax SVG rendering](https://docs.mathjax.org/en/latest/output/svg.html). Defaults to KaTeX.
```
can we link to Katex here as well?

eikowagenknecht replied:
Done

jackyzha0 commented on docs/features/Latex.md:
```diff
@@ -53,11 +54,6 @@ For example:
 - Incorrect: `I have $1 and you have $2` produces I have $1 and you have $2
 - Correct: `I have \$1 and you have \$2` produces I have \$1 and you have \$2
 
```
a common question we get about latex:

```
> [!warn]
> Due to limitations in the [underlying parsing library](https://github.com/remarkjs/remark-math),
> block math in Quartz requires the `$$` delimiters to be on newlines like above.
```

let's include this somewhere

eikowagenknecht replied:
Done (included in features/latex)

jackyzha0 commented on docs/plugins/HardLineBreaks.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/transformer
+---
+
+This plugin automatically converts single line breaks in Markdown text into hard line breaks in the HTML output. This plugin is not enabled by default.
```
```suggestion
This plugin automatically converts single line breaks in Markdown text into hard line breaks in the HTML output. This plugin is not enabled by default as this doesn't follow the semantics of actual Markdown but you may enable it if you'd like parity with [[Obsidian compatibility|Obsidian]].
```

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/FolderPage.md:
```diff
@@ -0,0 +1,22 @@
+
+This plugin generates index pages for folders, creating a listing page for each folder that contains multiple content files. See [[folder and tag listings]] for more information.
+
+Example: [[folder/advanced]]
```
this is a broken link isn't it? this should link to just `[[advanced/]]`

jackyzha0 replied:
(also note that links to folder pages should end with trailing slashes)

eikowagenknecht replied:
Fixed the link and included the note on the  `folder and tag listings` page

jackyzha0 commented on docs/plugins/Description.md:
```diff
@@ -0,0 +1,27 @@
+- as metadata for the HTML `head`
+- in FolderContent.tsx (TODO: What does it do there?)
+- in TagContent.tsx (TODO: What does it do there?)
+- in contentIndex.ts (TODO: What does it do there?)
```
- in folder content and tag content, if there is no main body content, it will use the description field as the text after the title and before the listing
- the content index uses this for the RSS feed (link to RSS here)

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/Description.md:
```diff
@@ -0,0 +1,27 @@
+- in TagContent.tsx (TODO: What does it do there?)
+- in contentIndex.ts (TODO: What does it do there?)
+
+If the frontmatter contains a `description` property, it is used (see [[authoring content#Syntax]]). Otherwise, the full text of the file is used.
```
not the full text, it will do a best-attempt to cut sentences until the target description length is reached

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/CreatedModifiedDate.md:
```diff
@@ -0,0 +1,23 @@
+- `priority`: The data sources to consult for date information. Highest priority first. Possible values are `"frontmatter"`, `"git"`, and `"filesystem"`. Defaults to `"frontmatter", "git", "filesystem"]`.
+
+> [!warning]
+> If you rely on git for dates, ensure `defaultDateType` is set to `modified` in `quartz.config.ts`.
```
```suggestion
> If you rely on `git` for dates, ensure `defaultDateType` is set to `modified` in `quartz.config.ts`.
```

jackyzha0 replied:
also maybe a warning here that 'filesystem' dates are not preserved when copying files for cloud hosting so if they want dates that persist after building in a CI/cloud hosting environment they should rely on git or frontmatter

eikowagenknecht replied:
Done, warning added

jackyzha0 commented on docs/plugins/CrawlLinks.md:
```diff
@@ -0,0 +1,27 @@
+
+This plugin accepts the following configuration options:
+
+- `markdownLinkResolution`: Defines the strategy for resolving Markdown paths, can be `"absolute"` (default), `"relative"` or `"shortest"`.
```
remark that these should match the user's obsidian configuration

jackyzha0 replied:
- absolute: path relative to root of content folder
- relative: path relative to file you are linking from
- shortest: name of the file, if that isn't enough to identify the file, use the full absolute path

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/CrawlLinks.md:
```diff
@@ -0,0 +1,27 @@
+This plugin accepts the following configuration options:
+
+- `markdownLinkResolution`: Defines the strategy for resolving Markdown paths, can be `"absolute"` (default), `"relative"` or `"shortest"`.
+- `prettyLinks`: When `true` (default), simplifies links by removing folder paths, making them more user-friendly.
```
e.g. folder/deeply/nested/note -> note

eikowagenknecht replied:
Done

jackyzha0 commented on docs/plugins/ContentPage.md:
```diff
@@ -0,0 +1,18 @@
+  - plugin/emitter
+---
+
+This plugin is a core component of the Quartz framework. It generates the HTML pages for each piece of Markdown content. It creates the full-page layout, including headers, footers, and body content, among others.
```
should leave a note here about configuring layout

eikowagenknecht replied:
Added link

jackyzha0 commented on docs/plugins/CNAME.md:
```diff
@@ -0,0 +1,22 @@
+
+This plugin emits a `CNAME` record that points your subdomain to the default domain of your site.
+
+If you want to use a custom domain name `quartz.example.com` for the site, then this is needed.
```
```suggestion
If you want to use a custom domain name like `quartz.example.com` for the site, then this is needed.
```

jackyzha0 commented on docs/plugins/Assets.md:
```diff
@@ -0,0 +1,20 @@
+  - plugin/emitter
+---
+
+This plugin emits all non-Markdown static assets (like images, scripts, stylesheets, etc.) from your vault content under the directory `content`.
```
```suggestion
This plugin emits all non-Markdown static assets in your content folder (like images, videos, HTML, etc).
```

jackyzha0 commented on docs/plugins/Assets.md:
```diff
@@ -0,0 +1,20 @@
+
+This plugin emits all non-Markdown static assets (like images, scripts, stylesheets, etc.) from your vault content under the directory `content`.
+
+Note that all static assets will then be accessible through its path on your generated site, i.e: `host.me/path/to/static.pdf`
```
also note that this respects `ignorePatterns`

eikowagenknecht replied:
Done

jackyzha0 commented on docs/configuration.md:
````diff
@@ -56,30 +56,48 @@ You can think of Quartz plugins as a series of transformations over content.
+],
+```
+
+To remove a plugin, you remove all occurrences of it in the `quartz.config.ts`.
````
```suggestion
To remove a plugin, you should remove all occurrences of it in the `quartz.config.ts`.
```

jackyzha0 commented on docs/configuration.md:
```diff
@@ -56,30 +56,48 @@ You can think of Quartz plugins as a series of transformations over content.
+
+To remove a plugin, you remove all occurrences of it in the `quartz.config.ts`.
+
+In addition, plugins may also have their own configuration settings that you can pass in. If you do not pass in a configuration, the plugin will use its default settings.
```
```suggestion
To customize plugins further, some plugins may also have their own configuration settings that you can pass in. If you do not pass in a configuration, the plugin will use its default settings.
```

jackyzha0 commented on docs/configuration.md:
```diff
@@ -56,30 +56,48 @@ You can think of Quartz plugins as a series of transformations over content.
+> Each node is modified by every transformer _in order_. Some transformers are position sensitive, so you may need to pay particular attention to whether they need to come before or after certain other plugins.
 
-Additionally, plugins may also have their own configuration settings that you can pass in. For example, the [[Latex]] plugin allows you to pass in a field specifying the `renderEngine` to choose between Katex and MathJax.
+To add a plugin, you insert a line with its function call, followed by a semicolon in the appropriate category. For example, to add the [[ExplicitPublish]] plugin (a [[tags/plugin/transformer|Transformer]], you would add the following line:
```
```suggestion
You should take care to add the plugin to the right entry corresponding to its plugin type. For example, to add the [[ExplicitPublish]] plugin (a [[tags/plugin/transformer|Transformer]], you would add the following line:
```

eikowagenknecht committed:
docs: update docs/plugins/RemoveDrafts.md

Co-authored-by: Jacky Zhao <j.zhao2k19@gmail.com>

eikowagenknecht committed:
docs: update docs/plugins/Assets.md

Co-authored-by: Jacky Zhao <j.zhao2k19@gmail.com>

eikowagenknecht committed:
docs: update docs/configuration.md

Co-authored-by: Jacky Zhao <j.zhao2k19@gmail.com>

eikowagenknecht committed:
docs: update docs/configuration.md

Co-authored-by: Jacky Zhao <j.zhao2k19@gmail.com>

eikowagenknecht committed:
docs: update docs/configuration.md

Co-authored-by: Jacky Zhao <j.zhao2k19@gmail.com>

eikowagenknecht committed:
docs: some updates

eikowagenknecht committed:
docs: work in review comments

eikowagenknecht commented:
Ok I hope I got all your comments worked in. From my side, this is good to be merged.

aarnphm commented:
Awesome thanks for the work.
